src/pkg/utils/utils.go

```diff
@@ -700,23 +702,26 @@ func ResolveImageName(distro, image, release string) (string, string, error) {
 	logrus.Debugf("Release: '%s'", release)
 
 	if distro == "" {
-		distro = distroDefault
+		distro = viper.GetString("general.distro")
 	}
 
-	if distro != distroDefault && release == "" {
-		return "", "", fmt.Errorf("release not found for non-default distribution %s", distro)
+	if release == "" && viper.InConfig("general.release") {
+		release = viper.GetString("general.release")
 	}
 
-	if release == "" {
-		release = releaseDefault
+	if distro != distroDefault && release == "" {
+		return "", "", fmt.Errorf("release not found for non-default distribution %s", distro)
 	}
 
-	if image == "" {
-		image = GetDefaultImageForDistro(distro, release)
+	if image == "" && viper.InConfig("general.image") {
+		image = viper.GetString("general.image")
+		release = ImageReferenceGetTag(image)
+	} else if image == "" {
+		image = getDefaultImageForDistro(distro, release)
```

Comment:
Umm... on second thoughts, this is also not quite correct.

The `--distro` and `--release` options on the command line should override the configuration file's `image` option.